### PR TITLE
use environment variable to select link target

### PR DESCRIPTION
**Pull request by BusyJay**
As discussed in pingcap/tikv#198 and considering the solutions other native crates have chosen (sfackler/rust-openssl#653), let's use environment variable to check link target.

### CODE REVIEW

**Comment by overvenus** [comment]
CC https://github.com/pingcap/kvproto/pull/198

**Comment by BusyJay** [comment]
ping

**Comment by siddontang** [comment]
How do we use now if we want to link static gRPC?

**Comment by BusyJay** [comment]
It will build and link to gRPC statically by default.

File: grpc-sys/build.rs [hunk 1]
```diff
@@ -12,146 +12,148 @@
...
+            panic!("unrecognize env var of {}: {:?}", name, s.to_string_lossy());
+        }
     }
 }
 
 fn main() {
     let mut cc = Build::new();
 
-    imp::build_or_link_grpc(&mut cc);
+    if get_env("GRPCIO_SYS_USE_PKG_CONFIG").map_or(false, |s| s == "1") {
```
Comment (siddontang) [inline grpc-sys/build.rs]: seem using 1 is to build grpc?

Comment (BusyJay) [reply grpc-sys/build.rs]: `GRPCIO_SYS_USE_PKG_CONFIG=1` means use `pkg-config` to search for pre-installed grpc package and link to it.

**Comment by siddontang** [comment]
LGTM

PTAL @overvenus

**Comment by BusyJay** [comment]
ping

**Pull request merged**